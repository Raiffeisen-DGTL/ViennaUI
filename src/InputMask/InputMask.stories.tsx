import React, {useCallback, useState } from 'react';
import { Story, Meta } from 'storybook';
import { InputMask, InputMaskProps } from './InputMask';

export default {
    title: 'Development/InputMask',
    component: InputMask,
} as Meta;

export const Overview: Story<InputMaskProps> = (args) => {
    const [value, setValue] = useState('');
    const changeHandler = useCallback((value) => setValue(value), []);
    return <InputMask maskOptions={{mask: Date}} value={value} onChange={changeHandler} {...args} />;
};
