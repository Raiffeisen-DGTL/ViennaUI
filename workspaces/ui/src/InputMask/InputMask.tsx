import React, { useCallback, useEffect, useRef, Ref } from 'react';
import IMask from 'imask';
import { IMaskProps, useMask } from 'vienna.react-use';
import { Input, InputProps } from '../Input/Input';

export interface Data {
    name?: string;
    value: string | number | Date;
    originalValue?: any;
}
export type InputMaskChangeEvent = (event: React.FormEvent<HTMLInputElement> | Event, data: Data) => void;

interface OmitedType extends Omit<InputProps, 'onChange'> {
    onChange?: InputMaskChangeEvent;
    showMask?: boolean;
    pipeRef?: Ref<{
        valueToMask: (string) => string;
        maskToValue: (string) => string;
        placeholder: (string) => string;
    }>;
}
export type InputMaskProps = OmitedType & IMaskProps;
export const InputMask: React.FC<InputMaskProps> = React.forwardRef(
    (props: InputMaskProps, ref: React.Ref<HTMLInputElement>) => {
        const {
            // start mask props
            mask,
            pipeRef,
            blocks,
            showMask = true,
            lazy = true,
            placeholderChar = '_',
            radix,
            thousandsSeparator,
            mapToRadix,
            min,
            max,
            scale,
            signed,
            normalizeZeros,
            padFractionalZeros,
            maxLength,
            from,
            to,
            prepare,
            format,
            parse,
            validate,
            autofix,
            overwrite,
            // end mask props
            value = '',
            name,
            onChange,
            ...attrs
        } = props;

        const maskOptions = {
            mask,
            blocks,
            lazy,
            autofix,
            overwrite,
            placeholderChar,
            radix,
            thousandsSeparator,
            mapToRadix,
            min,
            max,
            scale,
            signed,
            normalizeZeros,
            padFractionalZeros,
            maxLength,
            from,
            to,
            validate,
            prepare,
            format,
            parse,
        };

        const inputRef = useRef<any>();
        const maskRef = useRef<any>();
        const [valueToMask, maskToValue, placeholder] = useMask(maskOptions);

        useEffect(() => {
            if (inputRef.current) {
                if (typeof ref === 'function') {
                    ref(inputRef.current);
                } else if (ref && 'current' in ref) {
                    (ref as any).current = inputRef.current;
                }
            }

            return () => {
                if (typeof ref === 'function') {
                    ref(null);
                } else if (ref && 'current' in ref) {
                    (ref as any).current = null;
                }
            };
        }, [ref]);

        if (typeof pipeRef === 'function') {
            pipeRef({ valueToMask, maskToValue, placeholder });
        } else if (pipeRef && 'current' in pipeRef) {
            // eslint-disable-next-line
            (pipeRef as any).current = { valueToMask, maskToValue, placeholder };
        }

        const handleChange = useCallback(
            (e) => {
                if (typeof onChange === 'function') {
                    onChange(e, {
                        name: name,
                        value: maskRef.current.value,
                        originalValue: maskRef.current.unmaskedValue,
                    });
                }
            },
            [onChange, name]
        );

        useEffect(() => {
            if (inputRef.current) {
                maskRef.current = IMask(inputRef.current, maskOptions);
            }

            return () => {
                maskRef.current.destroy();
            };
        }, []);

        useEffect(() => {
            const inputRefCopy = inputRef;
            if (inputRef.current) {
                inputRef.current.addEventListener('input', handleChange);
            }

            return () => {
                if (inputRefCopy.current) {
                    inputRefCopy.current.removeEventListener('input', handleChange);
                }
            };
        }, [handleChange]);

        useEffect(() => {
            if (maskRef.current) {
                maskRef.current.updateOptions(maskOptions);
            }
        }, [maskOptions]);

        useEffect(() => {
            if (maskRef.current) {
                if (
                    value !== maskRef.current.value ||
                    (typeof value !== 'string' && maskRef.current.value === '' && !maskRef.current.el.isActive)
                ) {
                    maskRef.current.value = value === null ? '' : value;
                    maskRef.current.updateValue();
                }
            }
        }, [value]);

        return (
            <Input
                ref={inputRef}
                smartPlaceholder={showMask && value && placeholder(value)}
                name={name}
                maxLength={maxLength}
                {...attrs}
            />
        );
    }
);

InputMask.displayName = 'InputMask';
InputMask.defaultProps = {
    size: 'l',
    design: 'outline',
};

export default InputMask;
