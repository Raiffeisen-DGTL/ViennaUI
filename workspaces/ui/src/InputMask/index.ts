import InputMask from './InputMask';

export default InputMask;
export * from './InputMask';
export * from './Concrete/InputPhone/InputPhone';
export * from './Concrete/InputAccount/InputAccount';
export * from './Concrete/InputNumber/InputNumber';
export * from './Concrete/InputDate/InputDate';
export * from './Concrete/InputCard/InputCard';
export * from './Concrete/InputDateRange/InputDateRange';
export * from './Concrete/InputDigital/InputDigital';
