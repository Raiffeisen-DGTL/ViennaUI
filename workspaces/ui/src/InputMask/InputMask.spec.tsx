/* eslint-disable @typescript-eslint/no-misused-promises */
import React from 'react';
import InputMask, {
    InputPhone,
    InputDate,
    InputNumber,
    InputAccount,
    InputCard,
    InputDigital,
    InputDateRange,
    InputMaskProps,
} from './index';

it('InputMask', async () => {
    const mask = snapshot.render(<InputMask value='' />);

    expect(mask).toMatchSnapshot();
});

it('InputPhone', async () => {
    const mask = snapshot.render(<InputPhone />);

    expect(mask).toMatchSnapshot();
});

it('InputPhone /w focused', async () => {
    const mask = snapshot.mount(<InputPhone value='' />);
    const input = mask.find('input');

    input.simulate('focus');

    expect(mask).toMatchSnapshot();
});

it('InputPhone /w not complete', async () => {
    const mask = snapshot.render(<InputPhone value='4953' />);

    expect(mask).toMatchSnapshot();
});

it('InputPhone /w completed', async () => {
    const mask = snapshot.render(<InputPhone value='4959191242' />);

    expect(mask).toMatchSnapshot();
});

it('InputPhone /w onPaste', () => {
    function fn(e, data) {
        expect(JSON.stringify(data)).toBe(
            JSON.stringify({
                value: '+7 (495) 919-12-42',
                originalValue: '74959191242',
            })
        );
    }

    const mask = snapshot.mount(<InputPhone value='' onChange={fn} />);
    const input = mask.find('input');

    const event = (string) => ({
        nativeEvent: {
            clipboardData: {
                getData: () => string,
            },
        },
    });

    // base
    input.simulate('paste', event('4959191242'));
    // +7
    input.simulate('paste', event('+74959191242'));
    input.simulate('paste', event('+7 495 919 12 42'));
    input.simulate('paste', event('+7 495 919 1242'));
    input.simulate('paste', event('+7(495)9191242'));
    input.simulate('paste', event('+7(495) 9191242'));
    input.simulate('paste', event('+7(495)919-12-42'));
    input.simulate('paste', event('+7(495) 919-12-42'));
    // just 7
    input.simulate('paste', event('74959191242'));
    input.simulate('paste', event('7 495 919 12 42'));
    input.simulate('paste', event('7 495 919 1242'));
    input.simulate('paste', event('7(495)9191242'));
    input.simulate('paste', event('7(495) 9191242'));
    input.simulate('paste', event('7(495)919-12-42'));
    input.simulate('paste', event('7(495) 919-12-42'));
    // just 8
    input.simulate('paste', event('84959191242'));
    input.simulate('paste', event('8 495 919 12 42'));
    input.simulate('paste', event('8 495 919 1242'));
    input.simulate('paste', event('8(495)9191242'));
    input.simulate('paste', event('8(495) 9191242'));
    input.simulate('paste', event('8(495)919-12-42'));
    input.simulate('paste', event('8(495) 919-12-42'));
    // crazy (or old) gay style input
    input.simulate('paste', event('плюс 74959191242'));
    input.simulate('paste', event('плюс семь 4959191242'));
});

it('InputPhone /w onPaste custom mask', () => {
    function fn(e, data) {
        expect(JSON.stringify(data)).toBe(
            JSON.stringify({
                value: '+45 (11) 22-333',
                originalValue: '451122333',
            })
        );
    }

    const mask = snapshot.mount(<InputPhone value='' mask='+{45} (00) 00-000' onChange={fn} />);
    const input = mask.find('input');

    const event = (string) => ({
        nativeEvent: {
            clipboardData: {
                getData: () => string,
            },
        },
    });

    // +45
    input.simulate('paste', event('+451122333'));
    input.simulate('paste', event('+45 11 22 333'));
    input.simulate('paste', event('+45 11 22333'));
    input.simulate('paste', event('+45(11)22333'));
    input.simulate('paste', event('+45(11) 22333'));
    input.simulate('paste', event('+45(11)22-333'));
    input.simulate('paste', event('+45(11) 22-333'));
    // just 45
    input.simulate('paste', event('451122333'));
    input.simulate('paste', event('45 11 22 333'));
    input.simulate('paste', event('45 11 22333'));
    input.simulate('paste', event('45(11)22333'));
    input.simulate('paste', event('45(11) 22333'));
    input.simulate('paste', event('45(11)22-333'));
    input.simulate('paste', event('45(11) 22-333'));
    // crazy (or old) gay style input
    input.simulate('paste', event('my phone number is +45 11 22 333'));
});

it('InputPhone /w Eight mean Russia', () => {
    function fn(e, data) {
        expect(JSON.stringify(data)).toBe(
            JSON.stringify({
                value: '+456 (84) 56-112',
                originalValue: '4568456112',
            })
        );
    }

    const mask = snapshot.mount(<InputPhone value='' mask='+{456} (00) 00-000' onChange={fn} />);
    const input = mask.find('input');

    const event = (string) => ({
        nativeEvent: {
            clipboardData: {
                getData: () => string,
            },
        },
    });

    input.simulate('paste', event('8456112233'));
});

it('InputDate', async () => {
    const mask = snapshot.render(<InputDate value='22.02.1988' />);

    expect(mask).toMatchSnapshot();
});

it('InputDate /w new Date', async () => {
    const mask = snapshot.render(<InputDate value={new Date(1988, 1, 22)} />);

    expect(mask).toMatchSnapshot();
});

it('InputDate /w time', async () => {
    const mask = snapshot.render(<InputDate type='time' value='21:00' />);

    expect(mask).toMatchSnapshot();
});
it('InputDate /w datetime', async () => {
    const mask = snapshot.render(<InputDate type='datetime' value='22.02.1988 21:00' />);

    expect(mask).toMatchSnapshot();
});

it('InputDate /w not in range', async () => {
    const mask = snapshot.render(
        <InputDate value={new Date(2000, 5, 17)} min={new Date(2000, 5, 2)} max={new Date(2000, 5, 15)} />
    );

    expect(mask).toMatchSnapshot();
});

it('InputNumber', () => {
    const mask = snapshot.render(<InputNumber value='1000000' />);

    expect(mask).toMatchSnapshot();
});

it('InputAccount', async () => {
    const mask = snapshot.render(<InputAccount value='00000.810.0.000001' />);

    expect(mask).toMatchSnapshot();
});

it('InputCard', async () => {
    const mask = snapshot.render(<InputCard value='1234 1234 12' />);

    expect(mask).toMatchSnapshot();
});

it('InputDigital', async () => {
    const mask = snapshot.render(<InputDigital value='10.10' />);

    expect(mask).toMatchSnapshot();
});

it('InputDigital scale = 0', async () => {
    const mask = snapshot.render(<InputDigital value='10.10' scale={0} />);

    expect(mask).toMatchSnapshot();
});

it('InputDigital /w scale = 2', async () => {
    const mask = snapshot.render(<InputDigital value='10.10' scale={2} />);

    expect(mask).toMatchSnapshot();
});

it('InputDateRange', async () => {
    const mask = snapshot.render(<InputDateRange value='12.02.1930 - 22.0' />);

    expect(mask).toMatchSnapshot();
});

it('InputDateRange /w not in range', async () => {
    const mask = snapshot.render(
        <InputDateRange value={'01.06.2000 - 14.06.2000'} min={new Date(2000, 5, 2)} max={new Date(2000, 5, 12)} />
    );

    expect(mask).toMatchSnapshot();
});

test('InputDateRange w/ size', async () => {
    const sizes: InputMaskProps['size'][] = ['xs', 's', 'm', 'l', 'xl'];

    sizes.forEach(async (size) => {
        const snap = snapshot.render(<InputDateRange size={size} value='12.02.1930 - 22.0' />);
        expect(snap).toMatchSnapshot();
    });
});

test('InputDateRange w/ design', async () => {
    const designs: InputMaskProps['design'][] = ['outline', 'material'];

    designs.forEach(async (design) => {
        const snap = snapshot.render(<InputDateRange design={design} value='12.02.1930 - 22.0' />);
        expect(snap).toMatchSnapshot();
    });
});
