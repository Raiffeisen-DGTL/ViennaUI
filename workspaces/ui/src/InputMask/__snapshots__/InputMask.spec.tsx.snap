// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAccount 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          00000.810.0.000001_____
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        placeholder="_____.___._.___________"
        value="00000.810.0.000001"
      />
    </div>
  </div>
</label>
`;

exports[`InputCard 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          1234 1234 12__ ____
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        placeholder="____ ____ ____ ____"
        value="1234 1234 12"
      />
    </div>
  </div>
</label>
`;

exports[`InputDate /w datetime 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          22.02.1988 21:00
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="22.02.1988 21:00"
      />
    </div>
  </div>
</label>
`;

exports[`InputDate /w new Date 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          22.02.1988
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="22.02.1988"
      />
    </div>
  </div>
</label>
`;

exports[`InputDate /w not in range 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          17.06.2000
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="17.06.2000"
      />
    </div>
  </div>
</label>
`;

exports[`InputDate /w time 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          21:00
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="21:00"
      />
    </div>
  </div>
</label>
`;

exports[`InputDate 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          22.02.1988
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="22.02.1988"
      />
    </div>
  </div>
</label>
`;

exports[`InputDateRange /w not in range 1`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
          >
            01.06.2000 - 14.06.2000
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
          value="01.06.2000 - 14.06.2000"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange 1`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange w/ design 1`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange w/ design 2`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 lkLMMk"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 hEzoQJ"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 dyFbgS"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange w/ size 1`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 gPunPw"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 jNacye"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 hbWobN"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange w/ size 2`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 eablnU"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange w/ size 3`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 hmYQOg"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange w/ size 4`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDateRange w/ size 5`] = `
<div
  class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 dcSDdd"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Inputstyles__Box-hc8ttv-1 bzkyqR"
      >
        <div
          class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
        >
          <div
            class="Inputstyles__Placeholder-hc8ttv-0 xqPUL"
          >
            12.02.1930 - 22.0М.ГГГГ
          </div>
        </div>
        <input
          aria-invalid="false"
          class="Inputstyles__StyledInput-hc8ttv-2 QjOJw"
          value="12.02.1930 - 22.0"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`InputDigital /w scale = 2 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="10.10"
      />
    </div>
  </div>
</label>
`;

exports[`InputDigital 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="10.10"
      />
    </div>
  </div>
</label>
`;

exports[`InputDigital scale = 0 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="10.10"
      />
    </div>
  </div>
</label>
`;

exports[`InputMask 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value=""
      />
    </div>
  </div>
</label>
`;

exports[`InputNumber 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="1000000"
      />
    </div>
  </div>
</label>
`;

exports[`InputPhone /w completed 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <div
        class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
      >
        <div
          class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
        >
          +7 (495) 919-12-42
        </div>
      </div>
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="4959191242"
      />
    </div>
  </div>
</label>
`;

exports[`InputPhone /w focused 1`] = `
<InputPhone
  value=""
>
  <InputMask
    design="outline"
    mask="+{7} (000) 000-00-00"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholderChar="_"
    size="l"
    smartPlaceholder="+7 (___) ___-__-__"
    value="+7 ("
  >
    <Input
      design="outline"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
      onUpdated={[Function]}
      size="l"
      smartPlaceholder="+7 (___) ___-__-__"
      value="+7 ("
    >
      <InputWrapper
        active={true}
        design="outline"
        size="l"
      >
        <Inputstyles__Wrapper
          active={true}
          design="outline"
          size="l"
        >
          <label
            className="Inputstyles__Wrapper-hc8ttv-6 KDCUj"
            size="l"
          >
            <Inputstyles__Wrapped>
              <div
                className="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
              >
                <NativeInput
                  aria-invalid={false}
                  design="outline"
                  key=".1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onPaste={[Function]}
                  onUpdated={[Function]}
                  size="l"
                  smartPlaceholder="+7 (___) ___-__-__"
                  value="+7 ("
                >
                  <Inputstyles__Box>
                    <div
                      className="Inputstyles__Box-hc8ttv-1 bzkyqR"
                    >
                      <Inputstyles__PlaceholderWrapper>
                        <div
                          className="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
                        >
                          <Inputstyles__Placeholder
                            design="outline"
                            size="l"
                          >
                            <div
                              className="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
                              size="l"
                            >
                              <Inputstyles__Invisible>
                                <span
                                  className="Inputstyles__Invisible-hc8ttv-8 bCdWvB"
                                >
                                  +7 (
                                </span>
                              </Inputstyles__Invisible>
                              <span>
                                ___) ___-__-__
                              </span>
                            </div>
                          </Inputstyles__Placeholder>
                        </div>
                      </Inputstyles__PlaceholderWrapper>
                      <Inputstyles__StyledInput
                        aria-invalid={false}
                        design="outline"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onPaste={[Function]}
                        onScroll={[Function]}
                        size="l"
                        value="+7 ("
                      >
                        <input
                          aria-invalid={false}
                          className="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onPaste={[Function]}
                          onScroll={[Function]}
                          size="l"
                          value="+7 ("
                        />
                      </Inputstyles__StyledInput>
                    </div>
                  </Inputstyles__Box>
                </NativeInput>
              </div>
            </Inputstyles__Wrapped>
          </label>
        </Inputstyles__Wrapper>
      </InputWrapper>
    </Input>
  </InputMask>
</InputPhone>
`;

exports[`InputPhone /w not complete 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="4953"
      />
    </div>
  </div>
</label>
`;

exports[`InputPhone 1`] = `
<label
  class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
>
  <div
    class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
  >
    <div
      class="Inputstyles__Box-hc8ttv-1 bzkyqR"
    >
      <input
        aria-invalid="false"
        class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
        value="+7 ("
      />
    </div>
  </div>
</label>
`;
